tools/snp.sh

```diff
@@ -501,6 +505,33 @@ generate_guest_ssh_keypair() {
   ssh-keygen -q -t ed25519 -N '' -f "${GUEST_SSH_KEY_PATH}" <<<y
 }
 
+download_guest_os_image(){
+  local linux_distro=$(get_linux_distro)
+
+  # Set the guest OS image-cloud init URL, guest image basename based on the Host OS type
+  case ${linux_distro} in
+    ubuntu)
+      CLOUD_INIT_IMAGE_URL=${CLOUD_INIT_IMAGE_URL_UBUNTU}
+      IMAGE_BASENAME=${IMAGE_BASENAME_UBUNTU}
+      ;;
+    *)
+      >&2 echo -e "ERROR: ${linux_distro}"
+      return 1
+      ;;
+  esac
```

Comment:
Yes, Ubuntu will download Ubuntu guest image(debian based, which allows SNP guest kernel.deb to to be installed on debian guest)

RHEL, Fedora will download  Fedora image (rpm based, which allows SNP guest kernel rpm to be installed on rpm based guest OS)

Reply:
I see. I understand why it's been done this way. No need to address it in this PR but someday it would be nice to be able to mix and match.

Reply:
Yeah, we could use a standard single guest image(debian/rpm based) for all OS linux host distros provided if AMDESE/AMDSEV project can build a common SNP guest package format(e.g  deb) so that we can match the same SNP guest image(e.g ubuntu guest image) across different host linux distros